monkey/common/agent_events/file_encrypted_event.py

```diff
@@ -0,0 +1,70 @@
+from pathlib import PurePath, PurePosixPath, PureWindowsPath
+from typing import Any, Dict, Mapping, Tuple
+
+from pydantic import Field, validator
+
+from common import OperatingSystem
+
+from . import AbstractAgentEvent
+
+
+def _serialize_pure_path(pure_path: PurePath) -> Dict[str, str]:
+    # We generate data on either Windows or Linux and that data is processed
+    # on either OS which we can't sure where, so be need to serialize it as
+    # dictionary containing the path as string and the operating system
+    # which depends of the type of the path
+    serialized_os = (
+        OperatingSystem.LINUX.value
+        if isinstance(pure_path, PurePosixPath)
+        else OperatingSystem.WINDOWS.value
+    )
+    return {"path": str(pure_path), "os": serialized_os}
+
+
+class FileEncryptedEvent(AbstractAgentEvent):
+    """
+    An event that occurs when the Agent encrypts a file
+
+    Attributes:
+        :param file_path: Path of the encrypted file
+        :param success: Status of the file encryption
+        :param error_message: Message if an error occurs during file encryption
+    """
+
+    file_path: PurePath
+    success: bool
+    error_message: str = Field(default="")
+
+    class Config:
+        arbitrary_types_allowed = True
+        json_encoders = {PurePath: _serialize_pure_path}
+
+    @validator("file_path", pre=True)
+    def _file_path_to_pure_path(cls, v: Any) -> PurePath:
+        if isinstance(v, PurePath):
+            return v
+
+        if not isinstance(v, Mapping):
+            raise TypeError(f"Expected mapping but got {type(v)}")
+
+        path_string, os_string = FileEncryptedEvent._parse_path_dict(v)
+
+        if not isinstance(os_string, str):
+            raise TypeError("Operating system must be a string")
+
+        if os_string == OperatingSystem.LINUX.value:
+            return PurePosixPath(path_string)
+        elif os_string == OperatingSystem.WINDOWS.value:
+            return PureWindowsPath(path_string)
+
+        raise ValueError(f'"{os_string}" is not a valid operating system')
```

Comment:
We can avoid this boilerplate code by adding `source_os: OperatingSystem` to `FileEncryptedEvent` Then we can use [root validator](https://pydantic-docs.helpmanual.io/usage/validators/#root-validators) to cast it to whatever path we need based on OS

Reply:
Discussed on slack:

The OS is metadata about the path, not data about the event. The cost is some extra code, the benefit is the event isn't polluted with metadata.